rm/rm-node-updater/src/main/java/org/ow2/proactive/resourcemanager/updater/RMNodeUpdater.java

```diff
@@ -123,36 +145,185 @@ private static boolean makeNodeUpToDate() {
                     FileUtils.copyURLToFile(new URL(jarUrl), destination);
                     System.out.println("Download finished");
 
+                    cleanExpandDirectory(jarFile);
+
                     if (lock != null && lockFile != null) {
                         System.out.println("Releasing the lock on " + lockFile.getAbsoluteFile());
                         lock.release();
                     }
-
                     return true;
 
                 } catch (Exception e) {
                     System.err.println("Cannot download node.jar from " + jarUrl);
                     e.printStackTrace();
+                    return false;
                 }
+            } else {
+                return true;
             }
         } else {
-            System.out.println("No java property " + NODE_URL_PROPERTY +
-                               " specified. Do not check for the new version.");
+            throw new IllegalArgumentException("No java property " + NODE_URL_PROPERTY +
+                                               " specified. This property must be set when using " +
+                                               RMNodeUpdater.class.getSimpleName());
         }
+    }
 
-        return false;
+    /**
+     * Clean the directory used by One-Jar to expand libraries. After a node.jar update, we clean this directory to prevent jar conflicts
+     * @param jarFile name of the node jar file
+     * @throws IOException
+     */
+    private static void cleanExpandDirectory(String jarFile) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

Reply:
as mr sonar said....